lib/internal/crypto/random.js

```diff
@@ -119,6 +120,74 @@ function randomFill(buf, offset, size, cb) {
   _randomBytes(buf, offset, size, wrap);
 }
 
+// Largest integer we can read from a buffer.
+// e.g.: Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+const RAND_MAX = 281474976710655;
+
+function randomInt(min, max, cb) {
+  // randomInt(max, cb)
+  // randomInt(max)
+  if (typeof max === 'function' || typeof max === 'undefined') {
+    cb = max;
+    max = min;
+    min = 0;
+  }
+  const isSync = typeof cb === 'undefined';
+  if (!isSync && typeof cb !== 'function') {
+    throw new ERR_INVALID_CALLBACK(cb);
+  }
+  if (!NumberIsInteger(min)) {
+    throw new ERR_INVALID_ARG_TYPE('min', 'integer', min);
+  }
+  if (!NumberIsInteger(max)) {
+    throw new ERR_INVALID_ARG_TYPE('max', 'integer', max);
+  }
+  if (!(max > min)) {
+    throw new ERR_OUT_OF_RANGE('max', `> ${min}`, max);
+  }
+
+  // First we generate a random int between [0..range)
+  const range = max - min;
+
+  if (!(range <= RAND_MAX)) {
```

Comment:
I don't think this is sufficient. Both bounds should be safe integers, not just any integers. For example, `randomInt(Number.MAX_SAFE_INTEGER - 3, Number.MAX_SAFE_INTEGER + 3)` should not be valid, because the result is not guaranteed to be a safe integer, but the range is only `6`, and both bounds are integers, so it would currently pass validation.

On the other hand, the range does not necessarily have to be a safe integer (assuming an extension of the API to more than 48 bits). For example, `randomInt(-Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER)` should be valid, even though the range itself is not a safe integer (`2 ** 54`).

(This would also be much simpler if we didn't have an optional lower bound and always assumed zero. However, one could argue that not having a lower bound would expose users to the possibility of creating unsafe integers themselves.)

Reply:
Fixed in latest push.